src/main/java/org/folio/rest/transaction/AllOrNothingHandler.java

```diff
@@ -137,24 +148,42 @@ public void createTransaction(Transaction transaction) {
 
   }
 
-  protected Future<Void> verifyBudgetExistence(Transaction transaction){Promise<Void> promise = Promise.promise();
+  protected Future<Budget> getExistentBudget(Transaction transaction) {
+    Promise<Budget> promise = Promise.promise();
     Criteria criteria = getCriteriaByFieldNameAndValue("fundId", "=", transaction.getFromFundId());
     Criteria criteria1 = getCriteriaByFieldNameAndValue("fiscalYearId", "=", transaction.getFiscalYearId());
     Criterion criterion = new Criterion();
     criterion.addCriterion(criteria, "AND", criteria1);
-    getPostgresClient().get(BUDGET_TABLE, Transaction.class, criterion,true, reply -> {
+    getPostgresClient().get(BUDGET_TABLE, Budget.class, criterion,true, reply -> {
       if (reply.failed()) {
         handleFailure(promise, reply);
       } else if (reply.result().getResults().isEmpty()) {
         log.error(BUDGET_NOT_FOUND_FOR_TRANSACTION);
         promise.fail(new HttpStatusException(Response.Status.BAD_REQUEST.getStatusCode(), BUDGET_NOT_FOUND_FOR_TRANSACTION));
       } else {
-        promise.complete();
+        promise.complete(reply.result().getResults().get(0));
       }
     });
     return promise.future();
   }
 
+  private Future<Ledger> getExistentLedger(Transaction transaction) {
+    Promise<Ledger> promise = Promise.promise();
+    String query = "fund.id"+"="+ transaction.getFromFundId();
+    PgUtil.get(LEDGER_TABLE, Ledger.class, LedgerCollection.class, query , 0, 1, getOkapiHeaders(), getVertxContext(),
```

Comment:
Since we are making a call to GET ledger, don't we need to update permissions in the module descriptor?

Reply:
@javalkar-varun, this is storage module so there is no permissions to call DB.